pkg/util/converter/prom.go

```diff
@@ -0,0 +1,404 @@
+							Severity: data.NoticeSeverityError,
+							Text:     fmt.Sprintf("unable to parse key: %s in response body", l2Field),
+						})
+					}
+				}
+			}
+		default:
+			v := fmt.Sprintf("%v", iter.Read())
+			pairs = append(pairs, [2]string{l1Field, v})
+		}
+	}
+
+	return frame, pairs
+}
+
+func readMatrixOrVector(iter *jsoniter.Iterator) *backend.DataResponse {
+	rsp := &backend.DataResponse{}
+
+	for iter.ReadArray() {
+		timeField := data.NewFieldFromFieldType(data.FieldTypeTime, 0) // for now!
+		valueField := data.NewFieldFromFieldType(data.FieldTypeFloat64, 0)
+		valueField.Labels = data.Labels{}
+
+		for l1Field := iter.ReadObject(); l1Field != ""; l1Field = iter.ReadObject() {
+			switch l1Field {
+			case "metric":
+				iter.ReadVal(&valueField.Labels)
+
+			case "value":
+				t, v, err := readTimeValuePair(iter)
+				if err == nil {
+					timeField.Append(t)
+					valueField.Append(v)
+				}
+
+			// nolint:goconst
+			case "values":
+				for iter.ReadArray() {
+					t, v, err := readTimeValuePair(iter)
+					if err == nil {
+						timeField.Append(t)
+						valueField.Append(v)
+					}
+				}
+			}
+		}
+
+		valueField.Name = valueField.Labels["__name__"]
+		delete(valueField.Labels, "__name__")
+
+		frame := data.NewFrame("", timeField, valueField)
+		frame.Meta = &data.FrameMeta{
+			Type: data.FrameTypeTimeSeriesMany,
+		}
+		rsp.Frames = append(rsp.Frames, frame)
+	}
+
+	return rsp
+}
+
+func readTimeValuePair(iter *jsoniter.Iterator) (time.Time, float64, error) {
+	iter.ReadArray()
+	t := iter.ReadFloat64()
+	iter.ReadArray()
+	v := iter.ReadString()
+	iter.ReadArray()
+
+	tt := timeFromFloat(t)
+	fv, err := strconv.ParseFloat(v, 64)
+	return tt, fv, err
+}
+
+func readStream(iter *jsoniter.Iterator) *backend.DataResponse {
+	rsp := &backend.DataResponse{}
+
+	labelsField := data.NewFieldFromFieldType(data.FieldTypeString, 0)
+	labelsField.Name = "__labels" // avoid automatically spreading this by labels
+
+	timeField := data.NewFieldFromFieldType(data.FieldTypeTime, 0)
+	timeField.Name = "Time"
+
+	lineField := data.NewFieldFromFieldType(data.FieldTypeString, 0)
+	lineField.Name = "Line"
+
+	// Nanoseconds time field
+	tsField := data.NewFieldFromFieldType(data.FieldTypeString, 0)
+	tsField.Name = "TS"
+
+	labels := data.Labels{}
+	labelString := labels.String()
+
+	for iter.ReadArray() {
+
+		for l1Field := iter.ReadObject(); l1Field != ""; l1Field = iter.ReadObject() {
+			switch l1Field {
+			case "stream":
+				iter.ReadVal(&labels)
+				labelString = labels.String()
+
+			case "values":
+				for iter.ReadArray() {
+					iter.ReadArray()
+					ts := iter.ReadString()
+					iter.ReadArray()
+					line := iter.ReadString()
+					iter.ReadArray()
+
+					t := timeFromLokiString(ts)
+
+					labelsField.Append(labelString)
+					timeField.Append(t)
+					lineField.Append(line)
+					tsField.Append(ts)
+				}
+			}
+		}
+	}
+
+	frame := data.NewFrame("", labelsField, timeField, lineField, tsField)
```

Comment:
could we adjust this so that the first is the `timeField` and second is the `lineField` (rest does not matter)?

it makes my life easier when both metric-and-logs frames start with the timefield and then the valuefield.

Reply:
hmm.. after seeing https://github.com/grafana/grafana/blob/main/public/app/features/live/data/StreamingDataFrame.ts#L212-L213 i assume we want labels-field to be first :) ignore this one then, i'll update my code to handle this case.